src/test/java/eu/dissco/core/digitalspecimenprocessor/service/MidsServiceTest.java

```diff
@@ -39,13 +39,185 @@ void testCalculateMids(DigitalSpecimen digitalSpecimen, int midsLevel) {
   }
 
   private static Stream<Arguments> provideDigitalSpecimen() throws JsonProcessingException {
-    return Stream.of(Arguments.of(TestUtils.givenDigitalSpecimen(), 0),
+    return Stream.of(
+        Arguments.of(TestUtils.givenDigitalSpecimen(), 0),
         Arguments.of(MAPPER.readValue(MISSING_VALUE, DigitalSpecimen.class), 0),
         Arguments.of(MAPPER.readValue(WHITE_SPACE_VALUE, DigitalSpecimen.class), 0),
         Arguments.of(MAPPER.readValue(NULL_STRING_VALUE, DigitalSpecimen.class), 0),
-        Arguments.of(MAPPER.readValue(MIDS_1, DigitalSpecimen.class), 1));
+        Arguments.of(MAPPER.readValue(MIDS_1, DigitalSpecimen.class), 1),
+        Arguments.of(MAPPER.readValue(MISSING_LONGITUDE, DigitalSpecimen.class), 1),
+        Arguments.of(MAPPER.readValue(MISSING_QUALITATIVE_LOCATION, DigitalSpecimen.class), 1),
+        Arguments.of(MAPPER.readValue(UNKNOWN_SPECIMEN_TYPE, DigitalSpecimen.class), 1),
+        Arguments.of(MAPPER.readValue(BIO_LACKS_MEDIA, DigitalSpecimen.class), 1),
+        Arguments.of(MAPPER.readValue(BIO_MIDS_2, DigitalSpecimen.class), 2),
+        Arguments.of(MAPPER.readValue(MIDS_2, DigitalSpecimen.class), 2));
   }
 
+
+  private final static String MISSING_LONGITUDE = """
+      {
```

Comment:
I see why you instantiate the object like this, but reading from a hardcoded string seems like a bad precedent. Will this break if you just miss one space? Is there a better way to do this.

Reply:
Hmm the other way is to build ObjectNodes for this which would be really time consuming. I rather liked this approach as you can easily drop in an additional json and run it as a test. As long as it is valid Json it is fine. I pretty-formatted it so it is better readable.

Reply:
It definitely looks better than making a complex ObjectNode, good point.